src/MOIWrapper.jl

```diff
@@ -0,0 +1,357 @@
+export ECOSOptimizer
+
+using MathOptInterface
+const MOI = MathOptInterface
+const CI = MOI.ConstraintIndex
+const VI = MOI.VariableIndex
+
+const MOIU = MOI.Utilities
+
+const SF = Union{MOI.SingleVariable, MOI.ScalarAffineFunction{Float64}, MOI.VectorOfVariables, MOI.VectorAffineFunction{Float64}}
+const SS = Union{MOI.EqualTo{Float64}, MOI.GreaterThan{Float64}, MOI.LessThan{Float64}, MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives, MOI.SecondOrderCone, MOI.ExponentialCone}
+
+struct Solution
+    ret_val::Int
+    primal::Vector{Float64}
+    dual_eq::Vector{Float64}
+    dual_ineq::Vector{Float64}
+    slack::Vector{Float64}
+    objval::Float64
+end
+Solution() = Solution(0, Float64[], Float64[], Float64[], Float64[], NaN)
+
+mutable struct Data
+    m::Int
+    n::Int
+    IA::Vector{Int}
+    JA::Vector{Int}
+    VA::Vector{Float64}
+    b::Vector{Float64}
+    IG::Vector{Int}
+    JG::Vector{Int}
+    VG::Vector{Float64}
+    h::Vector{Float64}
+    objconstant::Float64
+    c::Vector{Float64}
+end
+
+mutable struct Cone
+    f::Int # number of linear equality constraints
+    l::Int # length of LP cone
+    q::Int # length of SOC cone
+    qa::Vector{Int} # array of second-order cone constraints
+    ep::Int # number of primal exponential cone triples
+    eqsetconstant::Dict{Int, Float64} # For the constant of EqualTo
```

Comment:
Same comment about not knowing what these fields do, given that the first fields are just ECOS's cone representation.